tools/clang/unittests/HLSL/CompilerTest.cpp

```diff
@@ -2320,6 +2321,12 @@ TEST_F(CompilerTest, CodeGenAttributeAtVertex) {
   CodeGenTestCheck(L"attributeAtVertex.hlsl");
 }
 
+TEST_F(CompilerTest, CodeGenConstantArrayInitializer) {
+  CodeGenTestCheck(L"function_call_initializer.hlsl");
```

Comment:
Leverage the batch tests. This should probably go into `batch\passes\your_pass_name`